source/server/options_impl.h

```diff
@@ -59,4 +67,23 @@ class OptionsImpl : public Server::Options {
   uint64_t max_stats_;
   uint64_t max_obj_name_length_;
 };
+
+/**
+ * Thrown when an OptionsImpl was not constructed because all of Envoy's work is done (for example,
+ * it was started with --help and it's already printed a help message) so all that's left to do is
+ * exit successfully.
+ */
+class NoServingException : public EnvoyException {
+public:
+  NoServingException() : EnvoyException("") {}
```

Comment:
Please put some text in the exception, even if it's just the name of the exception type or something